### ISSUE TITLE
Add legendgroup for annotations

### ISSUE THREAD

**Issue by ryandrew14** [plotly/plotly.js#4680]
One common use case I have run into for annotations is in multi-trace plots to have a series of annotations (one per trace) that each describe the same quality in their respective data. For use cases like this, where annotations are associated with specific traces, it would be awesome to be able to give them a `legendgroup` attribute so that they can be shown/hidden in the exact same situations as the traces they correspond to (turns out it's hard to replicate Plotly's native trace show/hide behavior with plotly_legendclick/doubleclick).

**Comment by ryandrew14** [comment]
@nicolaskruchten not sure who to go to, but are you open to a PR with this feature?

**Comment by nicolaskruchten** [comment]
Interesting idea! I think that would make sense to me, yes. @alexcjohnson thoughts? Better to tie it to a trace UID instead maybe?

**Comment by alexcjohnson** [comment]
Yeah, nice! I think I agree with @nicolaskruchten that tying to a trace UID would be cleaner. That way you can still group traces using `legendgroup` without needing to keep track of that on the annotations; and if we ever get around to [highlighting the hovered trace](https://github.com/plotly/plotly.js/issues/1847#issuecomment-312926089) that could also flow through to the annotation independent of `legendgroup`.

Also would be cool to do the same thing for shapes!

Now, what attribute name? Perhaps `showwithtrace`? `linkedtrace`? The attribute value would be a string, and the annotation would only be visible if a trace with matching `uid` attribute is also visible.

**Comment by ryandrew14** [comment]
Interesting! I have actually never come upon the trace uid -- could you point me to some documentation for how I'd discover and link it?

**Comment by gohsu** [comment]
Hi! Any update on this? Would be a really useful feature to have.

**Comment by ssaunders-hbk** [comment]
> I would still be interested by this feature. Currently using `plotly==5.24.1`.

I agree, this would definitely be a useful feature. I use R plotly, currently `plotly_4.10.4`.
